### DISCUSSION TITLE
Complex routes

### DISCUSSION THREAD

**Discussion by henk23** [nuxt/nuxt#16450] [Questions]
I want my router to respond to `/sitemap.xml`, `/sitemap-foo.xml`, `/sitemap-bar123-de_DE.xml` and other such constructs.
Since there does not seem to be a way to define routes like that (https://github.com/unjs/radix3/issues/11), I created a catchall route to handle all these different cases.

Problem now is: The catchall route catches all. :D Calling `/` or `/hello` results in 404. No way of displaying any of my other pages. I would love to have something like a `next()` function here to just disregard this route if it does not match.

Is this possible with Nuxt3?

## Reproduction
https://stackblitz.com/edit/nuxt-starter-3s4zs7?file=server%2Froutes%2F[sitemap].js

**Comment by TheAlexLichter** [comment]
Ideally you'd have sth. like `sitemap[slug].js`. Sadly you cannot mix static and dynamic parts in server routes yet. Follow along https://github.com/unjs/radix3/issues/11 ☺️

When using middleware, not returning anything should be equivalent to using `next()` in e.g. express.

**Reply by henk23** [reply]
Thanks @manniL, I agree, it "should be" but unfortunately it is not yet the case.

Also I'm not sure where to open a feature request: radix3, h3, nitro or nuxt3?

This issue, in combination with https://github.com/unjs/nitro/issues/630 kind of gives me a bad feeling about the current state of Nuxt3. "Release Candidate" sounds really hopeful, but at the moment it's not even possible to handle more than the most simple server routes. :/

**Reply by TheAlexLichter** [reply]
The linked issue should be sufficient for the feature request 😋

You can still handle these scenarios via middleware, Nitro is definitely capable of that ☺️

**Reply by henk23** [reply]
Cool, thanks for the tip, I'll look into middleware :)

**Reply by henk23** [reply]
Ah I remember why middleware did not seem a good fit for this. This is from the [docs](https://v3.nuxtjs.org/guide/directory-structure/server#server-middleware): 

> Middleware handlers should not return anything (nor close or respond to the request) and only inspect or extend the request context or throw an error.

So "should not" here is meant in an RFC way, it is possible, but be careful with it?
